xdsl/dialects/builtin.py

```diff
@@ -410,6 +410,15 @@ def from_float_and_width(value: float, width: int) -> FloatAttr[AnyFloat]:
 AnyFloatAttr: TypeAlias = FloatAttr[AnyFloat]
 
 
+@irdl_attr_definition
+class ComplexType(ParametrizedAttribute, TypeAttribute):
+    name: str = "complex"
+    element_type: ParameterDef[IntegerType | AnyFloat]
+
+    def __init__(self, element_type: IntegerType | AnyFloat) -> None:
+        super().__init__([element_type])
```

Comment:
How does this resolve to the proper parent init in python? I guess we want ParametrizedAttribute here, right?

Reply:
This just takes the immediate parent, so here `ParametrizedAttribute`.

Reply:
Wait there is also TypeAttribute, right?

Reply:
Oh god, that's a really good point, and I went in the rabbit hole of how these things works.
It works as is, but for clarity I'll change this to `ParametrizedAttribute` (we should probably do that everywhere), because apparently `super()` can be... quite weird sometimes.

Reply:
--> issue? :)

Reply:
Would making `TypeAttribute` abstract help?

Reply:
That depends on what abstract does with init. If it just makes it a crash, then this would not help as it might lead to crashes in weird circumstances. If it does something smarter, maybe in this case, but not in general, IMO.